src/main/java/com/sk89q/craftbook/mechanics/pipe/Pipes.java

```diff
@@ -317,9 +310,16 @@ private static boolean isValidPipeBlock(Block block) {
                         || SignUtil.isWallSign(block);
         }
     }
+    private String getPipeKey(Block block) {
+        return block.getWorld().getName() + ":" + block.getX() + "," + block.getY() + "," + block.getZ();
+    }
 
     private void startPipe(Block block, List<ItemStack> items, boolean request) {
-
+        String pipeKey = getPipeKey(block);
+        long currentTime = System.currentTimeMillis();
+        if(lastFailedAttempt.containsKey(pipeKey) && (currentTime - lastFailedAttempt.get(pipeKey) < 5000)) {
```

Comment:
Overall I don't feel this is something that can actually be added as is, without a configuration option.

If this does actually make sense (which is probably questionable), it'd have to be behind a configuration option as this will break most currently built systems that rely on any form of timing

Reply:
sorry didnt mean to add that part

Reply:
I'm happy for the original change of checking if the stack is valid before performing other logic, as that's a simple fast return optimisation - but the rest of it needs more thought and couldn't be merged as-is